test/es-module/test-typescript-commonjs.mjs

```diff
@@ -0,0 +1,166 @@
+import { spawnPromisified } from '../common/index.mjs';
+import * as fixtures from '../common/fixtures.mjs';
+import { match, strictEqual } from 'node:assert';
+import { test } from 'node:test';
+
+test('require a .ts file with explicit extension succeeds', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-strip-types',
+    '--eval',
+    'require("./test-typescript.ts")',
+    '--no-warnings',
+  ], {
+    cwd: fixtures.path('typescript/ts'),
+  });
+
+  strictEqual(result.stderr, '');
+  strictEqual(result.stdout, 'Hello, TypeScript!\n');
+  strictEqual(result.code, 0);
+});
+
+// TODO(marco-ippolito) This test should fail because extensionless require
```

Comment:
What's the plan on this?

Reply:
it a bit odd, because technically it shouldn't work because only `.js` files are allowed to be extensionless with the cjs loader. It should be looked into

Reply:
dealing with this after the first release which includes this feature will introduce a backwards-incompatible change, which, even though experimental, should be minimized I guess?

Reply:
yes this is priority, let's open an issue about it. At least the right behavior is well documented

Reply:
actualy I just figure out I'll prepare a fix

Reply:
@ChALkeR it's weird to talk to you like this, friendly reminder you're welcome to request to be re-added as a collaborator to the project now you're active again

Reply:
For posterity: https://github.com/nodejs/node/pull/54062